### PR TITLE
charts/victoria-metrics-k8s-stack: drops kubernetes high cardinality …

### PR DESCRIPTION
**Pull request by f41gh7**
…labels and metrics

### CODE REVIEW

**Comment by k1rk** [comment]
this is quite a breaking change. maybe lets do a flag for this behavior?

**Comment by f41gh7** [comment]
> this is quite a breaking change. maybe lets do a flag for this behavior?

I think it should be enabled by default and user may disable it if needed. Otherwise people will suffer from high disk usage for VictoriaMetrics indexDB

File: charts/victoria-metrics-k8s-stack/templates/servicemonitors/nodeexporter.yaml [hunk 1]
```diff
@@ -12,6 +12,10 @@ spec:
 {{- if empty (index .Values "prometheus-node-exporter" "vmServiceScrape" "spec") }}
   endpoints:
   - port: metrics
+    metricRelabelings:
```
Comment (jmenan) [inline charts/victoria-metrics-k8s-stack/templates/servicemonitors/nodeexporter.yaml:15]:
```suggestion
    metricRelabelConfigs:
```
metricRelabelings isn't supported by VMNodeScrape

File: charts/victoria-metrics-k8s-stack/values.yaml [hunk 2]
```diff
@@ -704,6 +704,15 @@ kubelet:
       insecureSkipVerify: true
       caFile: "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
     bearerTokenFile: "/var/run/secrets/kubernetes.io/serviceaccount/token"
+    # drop high cardinality label and useless metrics for cadvisor and kubelet
+    metricRelabelings:
```
Comment (jmenan) [inline charts/victoria-metrics-k8s-stack/values.yaml:708]:
```suggestion
    metricRelabelConfigs:
```
metricRelabelings isn't supported by VMNodeScrape

File: charts/victoria-metrics-k8s-stack/templates/servicemonitors/nodeexporter.yaml (hunk 1 above)
Comment (jmenan) [inline charts/victoria-metrics-k8s-stack/templates/servicemonitors/nodeexporter.yaml:15]:
```suggestion
    metricRelabelConfigs:
```

**Review by tenmozes — APPROVED** [review]

**Pull request merged**